eliot/testing.py

```diff
@@ -287,6 +288,28 @@ def checkForUnflushed():
 
 
 
+def capture_logging(assertion, *assertionArgs, **assertionKwargs):
+    """
+    Capture and validate all logging that doesn't specify a L{Logger}.
+
+    See L{validate_logging} for details on the rest of its behavior.
+    """
+    def decorator(function):
+        @validate_logging(assertion, *assertionArgs, **assertionKwargs)
+        @wraps(function)
+        def wrapper(self, *args, **kwargs):
+            logger = kwargs["logger"]
+            current_logger = _output._DEFAULT_LOGGER
+            _output._DEFAULT_LOGGER = logger
+
+            def cleanup():
+                _output._DEFAULT_LOGGER = current_logger
+            self.addCleanup(cleanup)
```

Comment:
And if not, perhaps the cleanup function should be registered *before* patching the global logger.